src/Integration/Telemetry/TelemetryDataRepository.cs

```diff
@@ -86,53 +84,49 @@ private void OnStorageFileChanged(object sender, FileSystemEventArgs e)
         private void EnsureFileExists(string filePath)
         {
             var directoryPath = Path.GetDirectoryName(filePath);
-            if (!directoryWrapper.Exists(directoryPath))
+            if (!fileSystem.Directory.Exists(directoryPath))
             {
-                directoryWrapper.Create(directoryPath);
+                fileSystem.Directory.CreateDirectory(directoryPath);
             }
-            if (!fileWrapper.Exists(filePath))
+            if (!fileSystem.File.Exists(filePath))
             {
                 WriteToXmlFile();
             }
         }
 
         private void WriteToXmlFile()
         {
-            TextWriter writer = null;
             var success = RetryHelper.RetryOnException(3, TimeSpan.FromSeconds(2),
                 () =>
                 {
-                    writer = fileWrapper.CreateText(StorageFilePath);
-                    this.ignoreFileChange = true;
-                    telemetrySerializer.Serialize(writer, Data);
-                    writer?.Flush();
+                    ignoreFileChange = true;
+                    var stringBuilder = new StringBuilder();
+                    telemetrySerializer.Serialize(new StringWriter(stringBuilder), Data);
```

Comment:
I couldn't mock CreateText, since it returns StreamWriter - which is not convertable to StringWriter.
Our original implementation casted it to TextWriter, which allowed the tests to run.
In this issue https://github.com/System-IO-Abstractions/System.IO.Abstractions/issues/392 they suggest to just switch to WriteAllText and ReadAllText, and I agree that they're simpler.